localization_manager/src/localization_manager_node.cpp

```diff
@@ -0,0 +1,145 @@
+        
+        // system_alert_topic_ protected member of CarmaLifecycleNode
+        system_alert_sub_ = create_subscription<carma_msgs::msg::SystemAlert>(system_alert_topic_, 1, 
+                                                                                std::bind(&LocalizationManager::systemAlertCallback, manager_.get(), std_ph::_1));
+        // Setup publishers
+        pose_pub_ = create_publisher<geometry_msgs::msg::PoseStamped>("selected_pose", 5);
+        state_pub_ = create_publisher<carma_localization_msgs::msg::LocalizationStatusReport>("localization_status", 5);
+        
+        // Create a publisher that will send all previously published messages to late-joining subscribers ONLY If the subscriber is transient_local too
+        rclcpp::PublisherOptions intra_proc_disabled; 
+        intra_proc_disabled.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable; // Disable intra-process comms for this PublisherOptions object
+
+        auto pub_qos_transient_local = rclcpp::QoS(rclcpp::KeepLast(1)); // A publisher with this QoS will store all messages that it has sent on the topic
+        pub_qos_transient_local.transient_local();  // A publisher with this QoS will re-send all (when KeepAll is used) messages to all late-joining subscribers 
+                                         // NOTE: The subscriber's QoS must be set to transient_local() as well for earlier messages to be resent to the later-joiner.
+        managed_initial_pose_pub_ = create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>("managed_initialpose", pub_qos_transient_local, intra_proc_disabled);
+
+        //Setup timer
+        pose_timer_ = create_timer(get_clock(), std::chrono::milliseconds(int(1/config_.pose_pub_rate * 1000)), 
+            std::bind(&LocalizationManager::posePubTick, manager_.get()));
+
+        // Return success if everything initialized successfully
+        return CallbackReturn::SUCCESS;
+
+    }
+
+    void Node::publishPoseStamped(const geometry_msgs::msg::PoseStamped& msg) const
+    {
+        pose_pub_->publish(msg);
+    }
+
+    void Node::publishStatus(const carma_localization_msgs::msg::LocalizationStatusReport& msg) const
+    {
+        state_pub_->publish(msg);
+    }
+
+    void Node::publishManagedInitialPose(const geometry_msgs::msg::PoseWithCovarianceStamped& msg) const
+    {
+        managed_initial_pose_pub_->publish(msg);
+    }
+
+    void Node::poseAndStatsCallback(const geometry_msgs::msg::PoseStamped::ConstPtr pose,
+                                     const autoware_msgs::msg::NDTStat::ConstPtr stats) const
+    {
+        try
+        {
+            manager_->poseAndStatsCallback(pose, stats);
+        }
+        catch (const std::exception& e)
+        {
+            RCLCPP_ERROR_STREAM(get_logger(), "Uncaught Exception in localization_manager. Exception: "<< e.what() );
```

Comment:
Why are we catching this exception and not allowing it to shutdown the system?

Reply:
If the issue is that it's not caught because of message_filters then we need to elevate it into the carma lifecycle node's error handling not just log it

Reply:
This is from the ros1 implementation https://github.com/usdot-fhwa-stol/carma-platform/blob/e7fe48309097169c29d811feb2456bbfce310e44/localization_manager/src/localizer_node.cpp#L49
If you think its an issue, i can change it here or log a bug and address it

Reply:
This is not equivalent because you are not passing the exception to the ros2 equivalent of handleException. That ROS1 function publishes a system_alert. You should call
handle_primary_state_exception(e); here to get the same behavior as in ROS1